backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlDatabase.java

```diff
@@ -179,9 +178,36 @@ public void init() { // setup default stuff
       // get the settings
       clearCache();
 
-      if (getSetting(Constants.IS_OIDC_ENABLED) == null) {
-        // use environment property unless overridden in settings
-        this.setSetting(Constants.IS_OIDC_ENABLED, String.valueOf(isOidcEnabled));
+      // oidc settings takes priority over env variables
+      String isOidcEnabledSetting = getSetting(Constants.IS_OIDC_ENABLED);
+      if (isOidcEnabledSetting != null) {
+        this.isOidcEnabled = Boolean.parseBoolean(isOidcEnabledSetting);
+      } else {
+        Object envSetting =
+            EnvironmentProperty.getParameter(Constants.MOLGENIS_OIDC_CLIENT_ID, null, STRING);
+        if (envSetting != null) {
+          this.setSetting(Constants.IS_OIDC_ENABLED, "true");
+          this.isOidcEnabled = true;
+        }
+      }
+      // check if OIDC settings are complete otherwise log error and set to false
+      if (this.isOidcEnabled) {
+        if (EnvironmentProperty.getParameter(Constants.MOLGENIS_OIDC_CLIENT_SECRET, null, STRING)
+            == null) {
```

Comment:
I think we agreed to not do this (my bad). So all OIDC parameters go via ENV except the OidcEnabled=true/false